avm/res/insights/private-link-scope/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,221 @@
+
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-insights.privatelinkscopes-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'iplswaf'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    logAnalyticsWorkspaceName: 'dep-${namePrefix}-la-${serviceShort}'
+    location: resourceLocation
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+var locationUpdated = toLower(replace(resourceLocation, ' ', ''))
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: 'global'
+    accessModeSettings: {
+      exclusions: [
+        {
+          ingestionAccessMode: 'PrivateOnly'
+          queryAccessMode: 'PrivateOnly'
+          privateEndpointConnectionName: 'thisisatest'
+        }
+      ]
+      ingestionAccessMode: 'Open'
+      queryAccessMode: 'Open'
+    }
+    scopedResources: [
+      {
+        name: 'scoped1'
+        linkedResourceId: nestedDependencies.outputs.logAnalyticsWorkspaceResourceId
+      }
+    ]
+    privateEndpoints: [
+      {
+        name: 'pe-${namePrefix}'
+        customNetworkInterfaceName: 'nic-pe-${namePrefix}'
+        subnetResourceId: nestedDependencies.outputs.subnetResourceId
+        privateDnsZoneResourceIds: [
+          nestedDependencies.outputs.privateDNSZoneResourceId
+        ]
+        tags: {
+          'hidden-title': 'This is visible in the resource name'
+          Environment: 'Non-Prod'
+          Role: 'DeploymentValidation'
+        }
+        roleAssignments: [
+          {
```

Comment:
Waf-aligned tests should be limited to only include properties related to best practices, leveraging psrule tests. For example role assignments for reader roles are normally not included

Reply:
@eriqua can you elaborate on _normally_ please? The PS Rule tests show success?
https://github.com/ahmadabdalla/bicep-registry-modules/actions/runs/7681415291/attempts/1#summary-20934593890

Is there a specific guidance for these tests?

Reply:
I have changed it btw. But keen on feedback on the above question please.

Reply:
In other words we try to limit test coverage of waf aligned tests to only properties related to waf. Everything else should be included in the max test. I'm sure PSRule checks are passing 🙂

Reply:
Same, e.g., for the long list of ipConfigurations. Are they all suggested to be waf compliant? Waf-aligned tests have been created in bulk in CARML as a duplicate of max tests. So we try to clean them up as part of the migration

Reply:
all IP configurations were required for a successful deployment

Reply:
So should the parameter be required? How would default test behave?